src/libpmemfile-posix/truncate.c

```diff
@@ -89,6 +93,11 @@ vinode_truncate(PMEMfilepool *pfp, struct pmemfile_vinode *vinode,
 			TX_SET_DIRECT(inode, mtime, tm);
 			TX_SET_DIRECT(inode, ctime, tm);
 		}
+
+		if (inode->allocated_space != allocated_space) {
+			TX_ADD_DIRECT(&inode->allocated_space);
```

Comment:
I'm not a huge fan of TX_SET* macros. IMO they hide too much. But I'm not going to block such changes.